components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineSession.kt

```diff
@@ -369,6 +377,11 @@ class SystemEngineSession(private val defaultSettings: Settings? = null) : Engin
         super.notifyObservers(block)
     }
 
+    private fun resetJSAlertData() {
+        alertJSCount = 0
+        shouldShowMoreJSDialogs = true
+    }
+
```

Comment:
I wonder if we can keep this out of the Session and do that in SystemEngineView (or a referenced helper) internally. In SystemEngineView we do have the callbacks from the clients where we may be able to reset certain things (instead of in the actions)? Is that feasible?

Reply:
Yeah,  initially I had everything on `SystemEngineView` but couldn't find callbacks for `goBack`,  `goForward` and `reload`, but I believe that resetting on `onPageStarted` will be sufficient, what do you think?

Reply:
Yeah, I assume that should be good enough.

Maybe another option would be to look at the URL in `onJsAlert()` and reset if the callback is invoked for a new URL?

Reply:
That could be a good approach too, but `url` that I'm getting from `onJsAlert` is  always **"about:blank"**. Is this an expected behavior?

Reply:
This doesn't sound right. Is this reproducible with a plain WebView? Hopefully we are not running into a WebView/Chrome bug :)

Reply:
The [web page that I was using for testing ](https://www.w3schools.com/jsref/tryit.asp?filename=tryjsref_alert) was triggering the alert from an `<iframe>` for this reason the url has the value of **"about:blank"** in other cases is returning the correct url.

Chrome
![image](https://user-images.githubusercontent.com/773158/50661227-737f7f00-0f70-11e9-8605-081357e3ebe5.png)

Fennec
![image](https://user-images.githubusercontent.com/773158/50661515-5dbe8980-0f71-11e9-935c-e98084f2ddc3.png)